src/memory-analyzer/memory_analyzer_main.cpp

```diff
@@ -0,0 +1,16 @@
+// Copyright 2018 Author: Malte Mues <mail.mues@gmail.com>
+#ifdef __linux__
+
+#include "memory_analyzer_parse_options.h"
+
+int main(int argc, const char **argv)
+{
+  memory_analyzer_parse_optionst parse_options(argc, argv);
+  return parse_options.main();
+}
+#else
+int main()
+{
+  throw "only supported on Linux platforms currently\n";
+}
```

Comment:
What are the challenges in making this cross platform?

Reply:
GDB is in general not supported out of the box on OSX and Windows. But without GDB, this code doesn't do anything meaningful.

Reply:
This is another argument ( IMHO ) for connecting to a remote GDB rather than spawning our own.